printer_mods/edwardyeeks/Decontaminator_Purge_Bucket_&_Nozzle_Scrubber/Macros/nozzle_scrub.cfg

```diff
@@ -0,0 +1,206 @@
+
+
+# These parameters define the size of your purge bucket. Update as necessary. If you decided to not enable purge, you can ignore 
+# this section. A visual reference is provided below. Note that orientation of parameters remain the same whether bucket is at rear 
+# or front.
+# 
+#                                     bucket_gap
+#                                      ← ---- →
+#                     __________________________________________
+#                    |                 |      |                 |      
+#                    |                 |      |                 |
+#  bucket_start (x)  |                 |______|                 |
+#                    |                 |      |                 |
+#                    |                 |      |                 |
+#                    |_________________|. . . |_________________|
+#                     ← ------------- →        ← ------------- →
+#                     bucket_left_width        bucket_right_width
+#  _______________________________________________________________________________________
+#                                    PRINTER FRONT
+# 
+## For V2.4 250mm build, uncomment below 
+#variable_bucket_left_width:    42
+
+## For V2.4 300mm build, uncomment below
+#variable_bucket_left_width:    67
+
+## For V2.4 350mm build, uncomment below
+#variable_bucket_left_width:    92                   
+
+# For V1.8, you may need to measure where your bucket start is and input into bucket_start. Otherwise, a value of 0 is for a default
+# installation of purge bucket at rear left.
+variable_bucket_start: 0
+
+
+###############################################################################################################################################
+###############################################################################################################################################
+
+### From here on, unless you know what you're doing, it's recommended not to change anything.
+
+###############################################################################################################################################
+###############################################################################################################################################
+
+# Placeholder. The variable will contain at random a number representing the left or right bucket.
+variable_bucket_pos:            1	        
+
+# This value is defaulted from brush location in CAD (rear left). Refer to diagram for visual reference.
+variable_brush_width:          52	
+
+# These values are defaulted from bucket geometry from CAD (rear left location). Refer to diagram for visual reference.
+variable_bucket_right_width:   40            
+variable_bucket_gap:           22	
+
+gcode:
+   # First, check if the axes are homed.
+   {% if "xyz" in printer.toolhead.homed_axes %}
+
+      ## Save the gcode state in this macro instance.
+      SAVE_GCODE_STATE NAME=clean_nozzle
```

Comment:
Since state is being saved, should probably run a `G90` here.  I've been testing this macro out and just ran into an issue where it tried to move out of range.

Reply:
You are right; good catch! Thanks.

Edward